mapproxy/client/wms.py

```diff
@@ -199,22 +199,31 @@ def __init__(self, request_template, http_client=None):
 
     def get_legend(self, query):
         resp = self._retrieve(query)
-        format = split_mime_type(query.format)[1]
```

Comment:
why did you remove this code? This is a change which may lead to problems because now you you are giving something like `image/png` to the `ImageSource` in line 206, while it was something like `png` before. Did you check if raster legends still work in different formats?

Reply:
when i stepped through the code with the debugger I found out that query.format equals 'png' (or jpeg or gif or json) and not 'image/png'. Therefore splitting the mime-type is useless (splitting does not return an error, it simply returns itself).

Reply:
i really don't mind resetting the code that splits the mimetype. done